### ISSUE TITLE
create match method/function to handle Maybe pattern match

### ISSUE THREAD

**Issue by VitorLuizC** [VitorLuizC/maybe#4]
Create a match function to enable _pattern match_ with Maybe none/some values.

```js
{
  none: () => ...,
  some: (value) => ...,
}
```

**Comment by VitorLuizC** [comment]
```ts
const size: Maybe<{ size: number, baseSize: number }> = await fetchSize();

const percentage = size.match({
  none: () => 0 + '%',
  some: ({ size, baseSize }) => (baseSize / size * 100) + '%'
});
```

**Comment by VitorLuizC** [comment]
Fixed and published on [`v2.0.0`](https://github.com/VitorLuizC/maybe/releases/tag/v2.0.0).
